packages/legacy/core/App/utils/helpers.ts

```diff
@@ -1012,16 +1013,27 @@ export const connectFromScanOrDeepLink = async (
 
   // TODO:(jl) Do we care if the connection is a deep link?
   logger.info(`Attempting to connect from scan or ${isDeepLink ? 'deeplink' : 'qr scan'}`)
-
   try {
-    const aUrl = processBetaUrlIfRequired(uri)
-    const receivedInvitation = await connectFromInvitation(aUrl, agent, implicitInvitations, reuseConnection)
-
-    if (receivedInvitation?.id) {
-      navigation.navigate(Stacks.ConnectionStack as any, {
-        screen: Screens.Connection,
-        params: { oobRecordId: receivedInvitation.id },
-      })
+    const parseResult = await parseInvitationUrl(uri)
+    if (!parseResult.success) {
+      throw new Error(`Primary error: ${parseResult.error}`)
+    }
+    const invitationData = parseResult.result
+    if (invitationData.type === 'didcomm') {
+      const aUrl = processBetaUrlIfRequired(uri)
+      const receivedInvitation = await connectFromInvitation(aUrl, agent, implicitInvitations, reuseConnection)
+
+      if (receivedInvitation?.id) {
+        navigation.navigate(Stacks.ConnectionStack as any, {
+          screen: Screens.Connection,
+          params: { oobRecordId: receivedInvitation.id },
+        })
+      }
+    } else if (invitationData.type === 'openid-credential-offer') {
+      //TODO: Impliment Navigation to display credential
+       // const record = await receiveCredentialFromOpenId4VciOffer({ agent: agent, uri: uri })
```

Comment:
We should avoid silent errors/dead-end, if it is not supported/implemented yet, it should throw an error.

Reply:
Will throw an error. This should be the next PR after implementing the base work

Reply:
resolved